Feature/166816215 user notifications

kevoese opened the pull request:
#### Description
Notification feature is implemented to allow users access to important information that they may have missed.
#### Type of change
Please select the relevant option

- [ ] Bug fix (non-breaking changes which fixes an issue)
- [x] New feature (non-breaking changes which adds functionality)
- [ ] Breaking change (fix or feature that could cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Chore

#### How Has This Been Tested?
please describe the test that you ran to verify your changes

- [ ] End to end 
- [x] Integration
- [X] Unit test

#### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generates no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

#### Pivotal Tracker
[#166816215](https://www.pivotaltracker.com/story/show/166816215)

andela-johia commented on controllers/users/index.js:
```diff
@@ -187,5 +191,64 @@ export default {
+        seen: false
+      }
+    });
+    if (notification) {
```
```
const notification = await db.Notification.findOne({
      where: {
        receiverId: id,
        id: notifyId
      }
    });
      if (notification) {
       const { seen } = notification;
       if (!seen) await notification.update({ seen: true })
      return res.status(200).send(notification)
}
```

Commit:
166816215-feature(notification): user can receive notification  …
  - add notifications folder
  - add methods for create article notification
  - add method for like and dislike notification
  - add methods for bulk notification
  - add pusher
  - Write test to cover feature
[Delivers #166816215]
